src/native/corehost/hostpolicy/deps_resolver.cpp

```diff
@@ -90,21 +90,22 @@ namespace
   // "asset_name" be part of the "items" paths.
   //
 void deps_resolver_t::add_tpa_asset(
-    const deps_resolved_asset_t& resolved_asset,
+    const deps_asset_t& asset,
+    const pal::string_t& resolved_path,
     name_to_resolved_asset_map_t* items)
 {
-    name_to_resolved_asset_map_t::iterator existing = items->find(resolved_asset.asset.name);
+    name_to_resolved_asset_map_t::iterator existing = items->find(asset.name);
     if (existing == items->end())
     {
         if (trace::is_enabled())
         {
             trace::verbose(_X("Adding tpa entry: %s, AssemblyVersion: %s, FileVersion: %s"),
-                resolved_asset.resolved_path.c_str(),
-                resolved_asset.asset.assembly_version.as_str().c_str(),
-                resolved_asset.asset.file_version.as_str().c_str());
+                resolved_path.c_str(),
+                asset.assembly_version.as_str().c_str(),
+                asset.file_version.as_str().c_str());
         }
 
-        items->emplace(resolved_asset.asset.name, resolved_asset);
+        items->emplace(asset.name, deps_resolved_asset_t(asset, resolved_path));
```

Comment:
Is this the place where copies were happening? 
(Making the argument an rvalue bypasses copying - Is that the idea of the change?)

Reply:
Yeah. Also figured we should avoid creating the `deps_resolved_asset_t` if we weren't going to need it (if the asset name is already in the map).